pkg/controllers/awsloadbalancercontroller/deployment.go

```diff
@@ -377,3 +395,63 @@ func indexedContainerEnv(envs []corev1.EnvVar) map[string]corev1.EnvVar {
 	}
 	return indexed
 }
+
+func hasSecurityContextChanged(current, desired *corev1.SecurityContext) bool {
+	if desired == nil {
+		return false
+	}
+
+	if current == nil {
+		return true
+	}
+
+	if desired.Capabilities != nil {
+		if current.Capabilities == nil {
+			return true
+		}
+		cmpCapabilities := cmpopts.SortSlices(func(a, b corev1.Capability) bool { return a < b })
+		if !cmp.Equal(desired.Capabilities.Add, current.Capabilities.Add, cmpCapabilities) {
+			return true
+		}
+
+		if !cmp.Equal(desired.Capabilities.Drop, current.Capabilities.Drop, cmpCapabilities) {
+			return true
+		}
+	}
+
+	if compareBoolPtr(current.RunAsNonRoot, desired.RunAsNonRoot) {
+		return true
+	}
+
+	if compareBoolPtr(current.Privileged, desired.Privileged) {
+		return true
+	}
+	if compareBoolPtr(current.AllowPrivilegeEscalation, desired.AllowPrivilegeEscalation) {
+		return true
+	}
+
+	if desired.SeccompProfile != nil {
+		if current.SeccompProfile == nil {
+			return true
+		}
+		if desired.SeccompProfile.Type != "" && desired.SeccompProfile.Type != current.SeccompProfile.Type {
+			return true
+		}
+	}
```

Comment:
same question as for the capabilities

Reply:
For both cases we don't want to set the value to `nil`. Ideally we will never have the desired as `nil`. If however in the first pass it gets created with `nil` it may be defaulted to something else due to webhooks. Subsequently, if we set it back to `nil` our controlller will end up constantly changing the value which is defaulted elsewhere.

Reply:
Yeah, I think it makes sense.
Then to be consistent, shouldn't we follow the same logic for the bool pointers (`RunAsNonRoot` etc.)? They are currently don't allow defaulting:
```go
		{
			name:      "desired RunAsNonRoot is nil",
			currentSC: &corev1.SecurityContext{RunAsNonRoot: pointer.BoolPtr(false)},
			desiredSC: &corev1.SecurityContext{},
			changed:   true,
		},
```